### PR TITLE
fix: return item() on closure to avoid LBFGS warning

### PR DESCRIPTION
**Pull request by cortexmancer**
While running attacks that use LBFGS to calculate the loss, torch keeps flushing a warning to the user:

```
/breaching/.venv/lib/python3.10/site-packages/torch/optim/lbfgs.py:457: UserWarning: Converting a tensor with requires_grad=True to a scalar may lead to unexpected behavior.
Consider using tensor.detach() first. (Triggered internally at /pytorch/torch/csrc/autograd/generated/python_variable_methods.cpp:835.)
  loss = float(closure())
```

This is related to issue [#160197](https://github.com/pytorch/pytorch/issues/160197) on torch. 

Using .item() while returning the loss in closure resolves the warning. 
Also, since an scalar float is now being returned, calls to item() and detach() was removed as well.

### CODE REVIEW

**Comment by JonasGeiping** [comment]
detach would be better to prevent to keep the tensor.

**Comment by JonasGeiping** [comment]
thanks!
